lib/streamlit/server/server.py

```diff
@@ -396,8 +419,51 @@ def _set_state(self, new_state):
         self._state = new_state
 
     @property
-    def is_ready_for_browser_connection(self):
-        return self._state not in (State.INITIAL, State.STOPPING, State.STOPPED)
+    async def is_ready_for_browser_connection(self) -> Tuple[bool, str]:
+        if self._state not in (State.INITIAL, State.STOPPING, State.STOPPED):
+            return True, "ok"
+
+        return False, "unavailable"
```

Comment:
I am reusing the HealthHandler class, which I made it async. The callback is now `await`-ed in the `get` method of the handler. Since the callback is `await`-ed, all the callback implementations should be async.

Or is async-await implementation in python different from other programming languages and I misunderstood how to use it?

Reply:
I missed that this was changed at the superclass level - yeah, that makes sense.